src/batch.js

```diff
@@ -1,28 +1,221 @@
+      }
+      return syncResults
+    })(),
+    ...asyncRequests.map(executeRequest.bind(null, config, route, event))
+  ])
 
   // return response
+  return {
+    statusCode: 200,
+    body:{ responses: results.map((result, index) => {
+      if (index === 0){
+        return result.map((result, index) => {
+          return {
+            id: syncRequests[index].id,
+            status: result.statusCode,
+            body: result.body ? JSON.parse(result.body) : undefined,
+            headers: result.headers
+          }
+        })
+      }
+      return {
+        id: asyncRequests[index - 1].id,
+        status: result.statusCode,
+        body: result.body ? JSON.parse(result.body) : undefined,
+        headers: result.headers
+      }
+    })},
+    headers:{}
+  }
 }
 
-function validateBatchRequest(body) {}
+async function executeRequest(config, route, event, request) {
+  //Parse url into pathParameters and multiValueQueryStringParameters formats
+  const urlAndQuerystring = request.url.split('?')
+  let queryStringParameters = {}
+  if (urlAndQuerystring.length > 1) {
+    const queryStringParams = urlAndQuerystring[1]
+    const keys = queryStringParams.split('&')
+    const queryStrings = keys.map((k) => k.split('='))
+
+    for (let queryString of queryStrings) {
+      if (queryStringParameters[queryString[0]]) {
+        queryStringParameters[queryString[0]].push(queryString[1])
+      } else {
+        queryStringParameters[queryString[0]] = [queryString[1]]
+      }
+    }
+  }
+
+  //Compose event to appear as native AWS event
+  let childEvent = { 
+    ...event,
+    httpMethod: request.method,
+    headers: {...normalizeRequestHeaders(request.headers), authorization: event.headers.authorization},
+    body: request.body,
+    path: request.url,
+    pathParameters: { 
+      proxy: urlAndQuerystring[0].split('/').splice(2).join('/') 
+    },
+    multiValueQueryStringParameters: queryStringParameters,
+    _json: true
+  }
+
+  //Allow consumers to define a custom handler 
+  if (config.handler){
+    return await config.handler(childEvent, {})
+  }
+
+  return await route(childEvent, {})
+}
+
+function normalizeRequestHeaders(reqHeaders = {}) {
+  return Object.keys(reqHeaders).reduce((headers, name) => {
+    headers[name.toLowerCase()] = reqHeaders[name]
+    return headers
+  }, {})
+}
+
+function splitAsyncRequests(orderedRequests) {
+  const asyncRequests = []
+  let syncRequests = []
+  for (let i = 0; i < orderedRequests.length; i++) {
+    if (orderedRequests[i].dependsOn.length > 0) {
+      syncRequests.push(...orderedRequests.splice(i))
+      break
+    }
+
+    if (!orderedRequests.some(req => req.dependsOn.includes(orderedRequests[i].id))) {
+      asyncRequests.push(orderedRequests[i])
+    } else {
+      syncRequests.push(orderedRequests[i])
+    }
+  }
+  return { asyncRequests, syncRequests }
+}
+
+//Depth-first topological sort
+function tSort(requests) {
```

Comment:
This isn't the best performing topological sort, but it is simple and easily allows us to detect circular dependsOn chains.